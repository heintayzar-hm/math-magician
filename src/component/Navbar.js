import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/home',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  const style = {
    nav: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-around',
      padding: '10px',
    },
    h1: { color: 'red', fontSize: '30px', flexBasis: '40%' },
    ul: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-around',
      width: '100%',
    },
  };
  return (
    <nav style={style.nav}>
      <h1 style={style.h1}>Math Magician</h1>
      <ul style={style.ul}>
        {links.map((link) => (
          <li key={link.id}>
            {' '}
            <NavLink
              to={link.path}
              style={({ isActive }) => ({ color: isActive ? 'red' : 'black' })}
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};
export default Navbar;
