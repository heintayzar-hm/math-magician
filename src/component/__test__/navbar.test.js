import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    // Arrange

    // Act
    const { container } = render(<HashRouter><Navbar /></HashRouter>);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the correct number of links', () => {
    // Arrange

    // Act
    const { container } = render(<HashRouter><Navbar /></HashRouter>);
    const navLinks = container.querySelectorAll('li');

    // Assert
    expect(navLinks.length).toBe(3);
  });
});
