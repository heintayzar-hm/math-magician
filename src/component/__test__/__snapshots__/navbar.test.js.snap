// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  style="display: flex; align-items: center; justify-content: space-around; padding: 10px;"
>
  <h1
    style="color: red; font-size: 30px; flex-basis: 40%;"
  >
    Math Magician
  </h1>
  <ul
    style="display: flex; align-items: center; justify-content: space-around; width: 100%;"
  >
    <li>
       
      <a
        class=""
        href="#/home"
        style="color: black;"
      >
        Home
      </a>
    </li>
    <li>
       
      <a
        class=""
        href="#/calculator"
        style="color: black;"
      >
        Calculator
      </a>
    </li>
    <li>
       
      <a
        class=""
        href="#/quote"
        style="color: black;"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
